### PR TITLE
Replace redux-thunk with redux-side-effect.

### PR DESCRIPTION
**Pull request by schuyler**
**TL;DR:** This replaces the somewhat complex system of thunk actions originally developed for importing from and exporting to the native contact store.

Instead, `loadNativeContacts()` is now just a regular static function that takes a `store` argument and dispatches the import actions to it, and the export of HelloAgain IDs on `toggleActive` is just called as a side effect from the `friends` reducer.

**More detail:**

Redux is great because it helps you separate functional concerns from side effects in user applications!

However, the Redux community seems to have no consensus on what to do with the side effects themselves. All of the add-on solutions are either wrapped in multiple layers of indirect execution and thus hard to test (thunks) or wrapped in multiple layers of declarative syntax and thus hard to code.

Eschewing additional libraries and just putting the side effects in a Redux store subscriber combines action at a distance with the problem of not being able to know what action was just triggered.

The `redux-side-effect` middleware just attaches a `.sideEffect()` method to every action. It takes a callback that's executed after the reducers finish, so that the side effects can be bundled with the "business logic" in the reducer, but in a way that's isolated and can be turned on/off as needed for testing various parts of the system properly.

Also, fortunately in practice, the friends reducer stays pure-_ish_, because the `toggleActive` helloAgainID export side effect is idempotent.

### CODE REVIEW

**Comment by obra** [comment]
+1

**Comment by schuyler** [comment]
Relevant to #66
